Source/Common/uri.h

```diff
@@ -0,0 +1,74 @@
+// Copyright (c) Microsoft Corporation
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+#pragma once
```

Comment:
is it worth making this public? We are using casablanca uri class right now and we could move to using this instead

Reply:
Maybe but its not a C API, so I'm leaning towards no